learning_resources_search/utils.py

```diff
@@ -0,0 +1,51 @@
+from learning_resources.hooks import get_plugin_manager
+from learning_resources_search.models import PercolateQuery
+
+
+def remove_child_queries(query):
+    """
+    Recursively removes 'has_child'/joins from a query
+    so it works with OpenSearch percolator
+    """
+    if isinstance(query, dict):
```

Comment:
If a user searches for resources with a text query like this:
http://localhost:8063/api/v1/learning_resources_search/?q=%22the%20content%20in%20this%20module?%22

That URL will return 1 xpro course that has contentfiles matching the text.  With the child join removed, it will likely return no results.  Not sure if that matters for percolate purposes though.

Reply:
this empty case is interesting.  I was thinking we would probably capture a bunch of these and follow up at some point.